testssl.sh

```diff
@@ -20336,26 +20357,147 @@ check_proxy() {
      fi
 }
 
+# Given the ASCII-HEX of a DER-encoded distinguished name, return the string
+# representation of the name.
+print_dn() {
+     local dn="$1"
+     local cert name
+     local -i len
+
+     # Use $OPENSSL to print the DN by creating a certificate containing the DN
+     # as the issuer and then having $OPENSSL print the issuer field in the
+     # resulting certificate.
+     cert="A003020102020100300A06082A8648CE3D040302${dn}301E170D3139303830353038333030305A170D3139303830353038333030305A30003019301306072A8648CE3D020106082A8648CE3D030107030200FF"
+     len=$((${#cert}/2))
+     if [[ $len -lt 128 ]]; then
+          cert="30$(printf "%02x" $len)$cert"
+     elif [[ $len -lt 256 ]]; then
+          cert="3081$(printf "%02x" $len)$cert"
+     else
+          cert="3082$(printf "%04x" $len)$cert"
+     fi
+     cert+="300A06082A8648CE3D040302030200FF"
+     len=$((${#cert}/2))
+     if [[ $len -lt 128 ]]; then
+          cert="30$(printf "%02x" $len)$cert"
+     elif [[ $len -lt 256 ]]; then
+          cert="3081$(printf "%02x" $len)$cert"
+     else
+          cert="3082$(printf "%04x" $len)$cert"
+     fi
+     name="$(asciihex_to_binary "$cert" | $OPENSSL x509 -issuer -noout -inform DER -nameopt RFC2253 2>/dev/null)"
```

Comment:
Short explanation of what RFC2253 means would be great or maybe a link to the section in the RFC

Reply:
I tried adding a comment, but I can't really point to a section of the RFC. RFC 2253 is "Lightweight Directory Access Protocol (v3): UTF-8 String Representation of Distinguished Names." So, the entire RFC is about how to represent a DN as a string.

Of course, we don't have to use the RFC 2253 (RFC 4514) string representation. RFC 2253 prints the attributes in the reverse of the order they actually appear in the DN:
> O = CAcert Inc., OU = http://www.CAcert.org, CN = CAcert Class 3 Root

leaving out "-nameopt RFC2253" would give us:

> CN=CAcert Class 3 Root,OU=http://www.CAcert.org,O=CAcert Inc.

Of course, there are many other choices of options to provide to "-nameopt". It's just a question of personal preference.